### ISSUE TITLE
Feel free to open up a PR!

### ISSUE THREAD

**Issue by amz-tools** [amz-tools/amazon-sp-api#58]
As the development of the Sellingpartner API is largely in progress by Amazon, there are quite a lot of changes and updates coming. We try to keep up with all new features/changes coming from Amazon, but if you find some things not working as expected or you have an idea of a feature that would be helpful and a good addition to the client you are very much welcome to open up a PR. We will try to review and include the PR inside the module as soon as possible, usually in a couple of days.

Some possible features for a PR could be i.e.:

- Full/better typesript support (see Issue #25)
- Re-Request `security_token` and/or `access_token` when throwing an invalid/malformed/revoked error (see Issue #57)
- "All in one" request and download report wrapper (`createReport`, `getReport`, `getReportDocument`, `download` all wrapped up in a convenient single call)
- Implement a "smart queue" on the basis of ratelimit and burst when requesting the same endpoint excessively (see Issue #46) 
- More tests

**Comment by jakeleventhal** [comment]
I've had a PR up for a few weeks now

**Comment by amz-tools** [comment]
@jakeleventhal Right, sorry for the slow integration time! We haven't really had a focus on the library in the last few months. However we will publish a new version in the next couple days that should include your PR as well.

**Comment by jakeleventhal** [comment]
@amz-tools added way more typings. including all report types. please take a look
